features/step_definitions/file_steps.coffee

```diff
@@ -4,18 +4,22 @@ path = require 'path'
 
 module.exports = ->
 
-  @Given /^I have no test files$/, (done) ->
-    done() # No op for readability
+  @Given /^my project has no test files$/, (done) ->
+    done()
 
 
-  @Given /^I have 1 (passing|failing) test$/, (status, done) ->
-    outputTestFile path.join(@tmpDir, 'example_spec.coffee'), passing: status is 'passing', done
+  @Given /^my project has a (passing|failing) test$/, (status, done) ->
+    outputTestFile path.join(@tmpDir, 'example_spec.coffee'),
+                   passing: status is 'passing',
+                   done
```

Comment:
My biggest problems are all about misalignment.
* When it happens the diff gets extra noise as realignment happens or things are left misaligned 
* My frustration of having to realign things (especially when a function definition starts at an odd number of spaces in). This is obviously compounded by my lack of an editor that can do this for me.

None of your reasons convinced me one was better then the other, and I held my ground because my way avoids these personal pain points (or pet peeves I guess)

Thinking more on understandability, I would prefer the arguments to be on a single line. As you pointed out its confusing to take in that line all at once. Thus I would like to propose we add some local variables to make the function call as easy to understand as possible.

I feel we went against the idea of unnecessary local variables in our recent projects (and I've passed that on). Thinking more on this now, I think local variables serve a powerful purpose as documentation.

For our example, I think using local variables makes the function call much simpler.
```coffee
@Given /^my project has a (passing|failing) test$/, (status, done) ->
  filePath = path.join(@tmpDir, 'example_spec.coffee')
  passing = status is 'passing'
  outputTestFile filePath, {passing}, done
```
It also makes the function interface look very weird. I designed that function under the assumption that a local variable wouldn't be used for passing. Merging the first two arguments together
```coffee
# spec/file_helpers.coffee
outputTestFile = ({filePath, passing}, done) ->
  fsExtra.outputFile filePath, testContent(passing), done

# file_steps.coffee
@Given /^my project has a (passing|failing) test$/, (status, done) ->
  filePath = path.join(@tmpDir, 'example_spec.coffee')
  passing = status is 'passing'
  outputTestFile {filePath, passing}, done
```
There is duplication between this and the another step as each has `passing = status is 'passing'. We can update outputTestFile to be able to handle that
```coffee
# spec/file_helpers.coffee
outputTestFile = ({filePath, passing, status}, done) ->
  if status then passing = status is 'passing'
  fsExtra.outputFile filePath, testContent(passing), done

# file_steps.coffee
@Given /^my project has a (passing|failing) test$/, (status, done) ->
  filePath = path.join(@tmpDir, 'example_spec.coffee')
  outputTestFile {filePath, status}, done
```
Overall I feel that was a nice cleanup of the code and made it nice and easy to understand for anyone looking at the code with fresh eyes. With named parameters we could now inline the variable again, though I'd prefer this form as its two very simple expressions versus a single more complex expression which would only same writing filePath once. 
````coffee
# file_steps.coffee
@Given /^my project has a (passing|failing) test$/, (status, done) ->
  outputTestFile {filePath: path.join(@tmpDir, 'example_spec.coffee'), status}, 
                 done
```